README.md

```diff
@@ -0,0 +1,705 @@
+    └── Complete job
+```
+
+### Inputs
+
+The workflows inputs are:
+- `pr` - boolean *'Publish to the Docker Official Library'* with the default value `true` (checked). The input indicates whether to create Pull Request to Docker with AlmaLinux *Definition File*.
+
+- `draft` - boolean '*Draft Pull Request*' with the default value `false` (not checked). The input indicates whether the [Pull Request is draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#draft-pull-requests) or it is ready to review.
+
+### Job: *version_major* *image_types* definition preparing
+
+Job iterates (using matrix) with AlmaLinux all `version_major`, and `image_types` (`default` and `minimal`). Multiple jobs run simultaneously for each of versions and for each of image types.
+
+#### Step: Checkout *container-images*, branch 'docker-library'
+
+The [actions/checkout@v4](https://github.com/actions/checkout/) checkouts *container-images* into branch 'docker-library'. The repository directory is located at `/home/runner/work/container-images/container-images`. All commits for the branch are checkout with `fetch-depth: 0`.
+
+#### Step: Checkout *official-images*, branch 'master'
+
+The [actions/checkout@v4](https://github.com/actions/checkout/) checkouts *container-images* into branch 'master'. The repository directory is located at `/home/runner/work/container-images/official-images`.
+
+That's your fork of [docker-library/official-images](https://github.com/docker-library/official-images) repository.
+
+#### Step: Get need data for the definition
+
+The step is written in bash. It reads the *# Tags:* string and the commit hash of `Containerfiles/${{ matrix.version_major }}/Containerfile.image_types` file changed by the **Build, test and push** workflow. `env.tags` and `env.last_commit` are exported. These data will be used to generate part of *Definition File*.
+
+#### Step: Render the definition
+
+The [chuhlomin/render-template@v1](https://github.com/marketplace/actions/render-template) generates from the `docker-library-definition.tmpl` using data (`env.tags`, `env.last_commit`, `matrix.version_major` and `matrix.image_types`) file `official-images/library/almalinux.version_major.image_types`. The file is a part of Docker Library *Definition File*.
+
+#### Step: Upload the definition for *version_major* *image_types*
+
+The step uses [actions/upload-artifact@v4](https://github.com/actions/upload-artifact) to store as artifact generated on previous step `official-images/library/almalinux.version_major.image_types`. The artifact is named against `version_major` and `image_type`, like `definition-version_major.image_types`.
+
+Artifacts are used to transfer files between different jobs of the same workflow. The artifact is zip archive of the file without file-path included.
+
+That is also possible to download artifacts via GitHub Action's web interface.
```

Comment:
The step uses [actions/upload-artifact@v4](https://github.com/actions/upload-artifact) to store an artifact generated in the previous step `official-images/library/almalinux.version_major.image_types`. The artifact is named against `version_major` and `image_type`, following the pattern: `definition-version_major.image_types`.

Artifacts are used to transfer files between different jobs of the same workflow. The artifact is a zip archive of the file without file-path included.

It is also possible to download artifacts via GitHub Action's web interface.